src/split-wrapper.js

```diff
@@ -0,0 +1,74 @@
+// split-wrapper.js
+var _ = require('underscore');
+
+var createLm = require('./create-lm');
+var createState = require('./create-state');
+var wrapper = require('./javascript-wrapper');
+var wrapperHelper = require('./wrapper-helper');
+
+/**
+ *  Define a default updater stub function to be used if the caller
+ * did not pass one in.  Simply returns whatever the original input was so it
+ * can be sent through the output port to the next component
+ *
+ * @param vnid_hash hash of vnids and data values 
+ */
+function defaultUpdater(vnid_hash) {
+    return vnid_hash;
+}
+
+function fRunUpdater(updater, updaterFormals, vni) {
+
+    return new Promise(function( resolve ) {
+        var results = wrapperHelper.executeUpdater(updater, 
+                                                   updaterFormals, 
+                                                   vni, 
+                                                   _.partial( wrapperHelper.updateOutputState, vni));
+        resolve(results);
+
+    }).then( function(results) {
+
+         if (! _.isUndefined(results)) {
+
+             if (!_.isObject(results)) {
+                 throw Error('Split wrapper requires a hash return from the updater');
+             }
+
+             // Update the output state with the content returned from the updater
+             var states = vni.inputStates();
+             var first = Object.keys(states)[0];
+             var groupLm = states[first].lm;  // Get the LM of first port - that will be the group LM
+
+             // Update output state but do NOT set the lm here - we don't want to send this
+             // Sending is done below on each of the vnids and they get their timestamp then
+             vni.outputState({data: results, groupLm: groupLm});  
+            
+             var hash = vni.outputState().data;
+             var lm = createLm(); // Create an lm for the new vnis we will be sending 
```

Comment:
Now that I think this through more, I think this lm could be set to the lm that we pulled from the input hash state (which at this point will be the same as the groupLm) instead of generating a new one.